sync_module/selective_sync/selective_sync.go

```diff
@@ -0,0 +1,163 @@
+// Package selectivesync implements a datasync manager.
+package selectivesync
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/services/datamanager"
+	"go.viam.com/rdk/services/vision"
+)
+
+// Model is the full model definition.
+var Model = resource.NewModel("selectivesync", "demo", "vision")
+
+func init() {
+	registration := resource.Registration[resource.Resource, *Config]{
+		Constructor: newSelectiveSyncer,
+	}
+	resource.RegisterComponent(generic.API, Model, registration)
+}
+
+// Config contains the name to the underlying camera and the name of the vision service to be used.
+type Config struct {
+	DataManager   string `json:"data_manager"`
+	Camera        string `json:"camera"`
+	VisionService string `json:"vision_service"`
+}
+
+// Validate validates the config and returns implicit dependencies.
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.DataManager == "" {
+		return nil, fmt.Errorf(`expected "data_manager" attribute in %q`, path)
+	}
+	if cfg.Camera == "" {
+		return nil, fmt.Errorf(`expected "camera" attribute in %q`, path)
+	}
+	if cfg.VisionService == "" {
+		return nil, fmt.Errorf(`expected "vision_service" attribute in %q`, path)
+	}
+
+	return []string{cfg.DataManager, cfg.Camera, cfg.VisionService}, nil
+}
+
+type visionSyncer struct {
+	resource.Named
+	actualDM      datamanager.Service
+	camera        camera.Camera
+	visionService vision.Service
+
+	cancelCtx  context.Context
+	cancelFunc func()
+	wg         sync.WaitGroup
+	logger     golog.Logger
+}
+
+func newSelectiveSyncer(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger,
+) (resource.Resource, error) {
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	v := &visionSyncer{
+		Named:      conf.ResourceName().AsNamed(),
+		logger:     logger,
+		cancelCtx:  cancelCtx,
+		cancelFunc: cancelFunc,
+	}
+	if err := v.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	v.wg.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer v.wg.Done()
+		v.startBackgroundProcess()
```

Comment:
So I don't think this will work as intended: since startBackgroundProcess is async, this will return immediately, even though that background process is still running. I think the easiest way to do this might be to just make startBackgroundProcess a synchronous function instead. Since this is already kicking everything off in a goroutine here, I think it should work pretty directly already if you just remove that `utils.PanicCapturingGo` at the top of `startBackgroundProcess`

Reply:
Ah thanks, updated!